client/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Route, RouteComponentProps, RouteProps, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+interface IProtectedRoute extends RouteProps {
+  Component: React.ComponentType;
+}
+
+const ProtectedRoute = ({ Component, ...rest }: IProtectedRoute): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  return (
+    <Route
+      render={(props: RouteComponentProps) => {
+        //for demonstrating performance issue
+        console.log('rendering...');
```

Comment:
Please get rid of these two lines :smiley:

Reply:
I left this there as part of the initial PR comment about the performance issue. In the second video that demonstrates the performance issue, all protected routes continuously render. I will remove it once the issue has been resolved.